### PR TITLE
Fix tax strategy strict validation

### PR DESCRIPTION
**Pull request by lkostrowski**
Fixes too strict validation of Saleor payload, causing app returning 500 instead gracefully ignoring the event

### CODE REVIEW

File: apps/avatax/src/app/api/webhooks/order-confirmed/route.ts [hunk 1]
```diff
@@ -152,25 +152,26 @@ const handler = orderConfirmedAsyncWebhook.createHandler(async (_req, ctx) => {
...
+            errorReason: "Order channel does not use the tax app strategy",
           })
             .mapErr(captureException)
             .map(logWriter.writeLog);
 
           span.setStatus({
             code: SpanStatusCode.OK,
-            message: "Failed to commit transaction in AvaTax: order has flat tax rates strategy",
+            message:
+              "Failed to commit transaction in AvaTax: order channel does not use the tax app strategy",
```
Comment (stmpn) [inline apps/avatax/src/app/api/webhooks/order-confirmed/route.ts:169]: This might be just me being nitpicky, but `Failed to...` in conjunction with `SpanStatusCode.OK` seems confusing, maybe `Skipped commit...`?

File: apps/avatax/src/modules/saleor/order-confirmed/event.test.ts
Comment (stmpn) [inline apps/avatax/src/modules/saleor/order-confirmed/event.test.ts:1]: We are missing test to confirm we return 2XX instead of 5XX when strategy is `null`

**Review by stmpn — APPROVED** [review]
Two suggestions, but not a deal breakers. Would like to see them adressed, but if you decide they are not important, I am fine with that.
